app/models/mixins/aggregation_mixin/methods.rb

```diff
@@ -39,13 +39,10 @@ def all_storages
     end
 
     def aggregate_hardware(from, field, targets = nil)
-      from      = from.to_s.singularize
-      select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
-      targets ||= send("all_#{from}_ids")
-      targets   = targets.collect(&:id) unless targets.first.kind_of?(Integer)
-      hdws      = Hardware.where("#{from}_id" => targets).select(select)
-
-      hdws.inject(0) { |t, hdw| t + hdw.send(field).to_i }
+      from_singular = from.to_s.singularize
+      targets ||= send("all_#{from}")
+      targets   = targets.collect(&:id) if targets.kind_of?(Array) && !targets.first.kind_of?(Integer)
```

Comment:
The `collect(&:id)` seems like something we could build into the query.  I figured you would have done this yourself, though, so is there a reason why you didn't?

Reply:
@NickLaMuro This is focused around when `targets.kind_of?(Array)`, which is not a typical use case. I left as is, but did ensure it was an `Array` (so we didn't kick off the unnecessary query when a scope was passed in)

I agree, `select(:id)` is better. This tries very hard to only execute that mapping if it is not from the db.